lib/Client.js

```diff
@@ -567,8 +581,22 @@ Client.prototype._onclose = function(err) {
   if (!err)
     this.emit('end');
   this.emit('close');
+  /**
+   there was a serious memory leak issue when using Client in a pool setup. 
+   if a client was used once and then close()ed the _handle was causing Client to never be garbage collection correctly. 
+   so the bindings was leaking around xMB everytime. 
+   the details around this is documented here 
+   https://github.com/mscdex/node-mariasql/pull/130 
+   and
+   https://github.com/mscdex/node-mariasql/pull/133
+  **/
+  process.nextTick(this._clearHandle);
```

Comment:
This won't work since the `this` context will be lost when `_clearHandle()` is called.